src/az390.java

```diff
@@ -4400,6 +4455,22 @@ private void process_bal_op(){
        get_hex_vreg(2);        // V2
        skip_comma();
        get_hex_vreg(3);        // V3
+	   if (   tz390.op_code[bal_op_index].substring(0,4).equals("E67D")){
+		 // VRRj map oo1230m00xoo DSH #414
+         get_hex_zero(1);
+         if (!bal_abort && exp_next_char(',')){
+    	 	skip_comma();
+    		get_hex_reg();   // M4
+         } else {
+			if ((tz390.op_code[bal_op_index].length() == 5)) {
```

Comment:
Might want to add a comment explaining what's being checked here. Not at all clear to me.

Reply:
Agreed. Pretty convoluted code. PoP says masks are specified as M6,M5,M4 which is at odds with the comments.
It  look like the program is testing the mask operands one by one, but M6 and M5 are obligatory. Or do they default to zero when omitted? I tried to understand in order to add comments - but I'm worried I'd get it wrong.